src/services/search/typeSearch.ts

```diff
@@ -75,14 +72,17 @@ export const fetchTypes = async (
   }
 }
 
-export const fetchTypeFacets = async (
-  query: SearchTypeQueryParams = {},
-  fields: string[] = [],
-  options: RequestOptions
-): Promise<WithValidationError<Facets>> => {
+export const fetchTypeFacets = async ({
+  query = {},
+  fields = [],
+  options = {},
+}: {
+  query?: SearchTypeQueryParams
```

Comment:
Hmm interesting, I have in mind that I get linter errors in the project when marking properties that default to something as optional.

Reply:
What case are you talking about? That seems like an error in the linter rules TBH, can you check if that's still the case and create a ticket if so?

Reply:
I might be of or it behaves differently when using it as part of an object..
Had that one in mind
![image](https://user-images.githubusercontent.com/3371760/96706551-046a5900-1397-11eb-8ab5-e46c2e5bcd90.png)

Reply:
Okay, that's a different case.
In your example you have:
```
const foo = (param: string = "string") => ...
```
which is an argument of a function,
whereas here we have:
```
const foo = ({ bar = "string" }: { bar?: string })  => ....
```
which instantiates a field of the object.
It makes sense to error out in the first case since by assigning default you create two call signatures:
```
foo(param: string)
foo()
```
whereas in the second case it does not touch call signatures, only ensures that the field has a default value.

Reply:
Thanks for the explanation, thought somehow it would compare the call sinature of the argument the same way....